pkg/kubelet/dockershim/exec.go

```diff
@@ -128,6 +130,13 @@ func (*NsenterExecHandler) ExecInContainer(client libdocker.Interface, container
 	if exitErr, ok := cmdErr.(*exec.ExitError); ok {
 		return &utilexec.ExitErrorWrapper{ExitError: exitErr}
 	}
+
+	// since native methods of command do not provide timeout mechanisms
+	// it is considered safe to wait until command ends and have a judgement of timeout after that
+	if timeout >= 0 && time.Now().Unix()-start >= int64(timeout.Seconds()) {
```

Comment:
i think it would be better to get a duration in here... time.Before or time.Since

Reply:
good catch